core/src/main/java/org/infinispan/remoting/transport/impl/SingleTargetRequest.java

```diff
@@ -83,12 +99,10 @@ private T addResponse(Address sender, Response response) {
    protected void onTimeout() {
       // The target might be null
       String targetString = null;
-      synchronized (responseCollector) {
-         if (requestTracker != null) {
-            requestTracker.onTimeout();
-            targetString = requestTracker.destination().toString();
-         }
-
+      final RequestTracker tmp = requestTracker;
+      if (tmp != null) {
+         tmp.onTimeout();
```

Comment:
Technically it is possible to have a metric mismatch now where under concurrent timeout with completion that we report a timeout or completion, but the opposite actually applied.

This case is extremely unlikely though and probably isn't a big concern for this PR.